src/cast_core/src/CrudActor.cpp

```diff
@@ -1132,6 +1135,87 @@ std::string getDbName(const PhaseContext& phaseContext) {
     return phaseDb ? *phaseDb : *actorDb;
 }
 
+enum Units { NANOSECOND, MICROSECOND, MILLISECOND, SECOND, MINUTE, HOUR };
+// represent a random delay
+class Delay {
```

Comment:
I nearly made this a ValueGenerator in DocumentGenerators, but I wasn't sure it made sense for generating bson, so I just put it here.

Reply:
It might make sense for this to be a TypeGenerator<Duration> and move to Generators. I don't think it makes sense to be Generator<>

Reply:
This could be viewed as a TypeGenerator<Duration>. where Duration is using Duration = typename TimeSpec::ValueT;  It would be a little bit of work, but I could probably make that a thing if desired. 

So, it is using TimeSpec.